src/hooks/useActiveDescendantListFocus/__tests__/useActiveDescendantListFocus.jest.js

```diff
@@ -78,37 +112,153 @@ function runListUnitTest(isHorizontal) {
   });
 }
 
+const combineFeatures = features => {
+  const combinations = [];
+  // all closed
+  const allClosedComb = {};
+  for (let feature of features) {
+    allClosedComb[feature] = false;
+  }
+  combinations.push(allClosedComb);
+
+  // all open
+  const allOpenComb = {};
+  for (let feature of features) {
+    allOpenComb[feature] = true;
+  }
+  combinations.push(allOpenComb);
+
+  // one open all other closed
+  for (const feature of features) {
+    const oneOpen = {};
+    oneOpen[feature] = true;
+    for (const otherFeature of features) {
+      oneOpen[otherFeature] = false;
+    }
+    combinations.push(oneOpen);
+  }
+
+  return combinations;
+};
+
 describe("useActiveDescendantListFocus", () => {
   afterEach(() => {
     element.remove();
     cleanup();
   });
 
-  describe("Vertical list", () => {
-    const isHorizontal = false;
-    runListUnitTest(isHorizontal);
+  const features = {
+    isHorizontal: { yes: "Horizontal", no: "Vertical" },
+    defaultVisualFocusFirstItem: { yes: "KeepOpen", no: "" }
+  };
+
+  const featureCombinations = combineFeatures(Object.keys(features));
+  for (const featureComb of featureCombinations) {
+    describe(`${Object.entries(featureComb)
+      .map(([feat, value]) => (value ? features[feat].yes : features[feat].no))
+      .filter(b => b)
+      .join(", ")} list`, () => {
+      runListUnitTest(featureComb);
+    });
+  }
+
+  describe("defaultVisualFocusFirstItem option", () => {
+    const defaultVisualFocusFirstItem = true;
+    it("should focus same item after item changed", async () => {
+      const onItemClick = jest.fn();
+      const { result, rerender } = renderHookForTest({ onItemClick, defaultVisualFocusFirstItem });
+      const moveForwardKey = "{arrowDown}";
+      act(() => {
+        // set focus on the list's element which in charge on natural focus element
+        element.focus();
+      });
+      for (let idx = 0; idx < 3; idx++) {
+        act(() => {
+          userEvent.keyboard(moveForwardKey);
+        });
+      }
+
+      const beforeIndex = result.current.visualFocusItemIndex;
+      const beforeId = result.current.visualFocusItemId;
+
+      const removeItems = 2;
+      const itemsAfterRemove = ITEM_IDS.slice(removeItems);
+      rerender({ itemsIds: itemsAfterRemove });
+
+      expect(result.current.visualFocusItemId).toEqual(beforeId);
+      expect(result.current.visualFocusItemIndex).toEqual(beforeIndex - removeItems);
+    });
+
+    it("should focus correct item on focus", async () => {
+      const onItemClick = jest.fn();
+      const { result } = renderHookForTest({ onItemClick, defaultVisualFocusFirstItem });
+
+      act(() => {
+        // set focus on the list's element which in charge on natural focus element
+        element.focus();
+      });
+
+      expect(result.current.visualFocusItemIndex).toEqual(0);
+    });
+
+    it("should focus first selectable one only", async () => {
```

Comment:
it's a copy paste of the tests for is selectable that already in the general unit tests no?
I think we can delete it?